app/models/hub_edos/common/reference/descriptor.rb

```diff
@@ -0,0 +1,31 @@
+module HubEdos
+  module Common
+    module Reference
+      # maps an abstract or abbreviated code to a set of descriptions
+      class Descriptor
+        # a short string used as a placeholder for the associated description	string
+        attr_accessor :code
+
+        # the full description implied by the code
+        attr_accessor :description
+
+        # a longer, more official version of the description
+        attr_accessor_with_lower_camelized_alias :formal_description
+
+        def initialize(attrs = {})
+          attrs.each do |key, value|
+            send("#{key}=", value) if respond_to?("#{key}=")
+          end
+        end
+
+        def as_json(options={})
+          {
+            code: code,
+            description: description,
+            formalDescription: formal_description,
+          }.compact
```

Comment:
I think it may be easier for the UI to handle a null value for a property of an object that is included rather than requiring it check for the presence and value. In less words, maybe we shouldn't compact this?

Reply:
This is a general purpose class, in some contexts the formalDescription isn't present, so it's probably best to just not include it if we don't receive it.